AutonomousDemo/DeterminePath.cs

```diff
@@ -9,24 +9,34 @@ namespace AutonomousDemo
     public class DeterminePath
     {
         Road road = new Road(27);
-        
+
         public Velocity CalculateDriveVelocity(Vehicle vehicle)
         {
             if (vehicle.VehicleMotion.CurrentVelocity.Speed < road.SpeedLimit)
             {
-                return vehicle.VehicleMotion.AccelerateVelocityPerSecond(vehicle.VehicleMotion.CurrentVelocity);
+                return vehicle.VehicleMotion.Accelerate(vehicle.VehicleMotion.CurrentVelocity);
             }
-                return vehicle.VehicleMotion.CurrentVelocity;
+            return vehicle.VehicleMotion.CurrentVelocity;
+        }
+
+        public Velocity StoppingDistance(Vehicle vehicle, double destinationDistance)
+        {
+            var stoppingDistance = destinationDistance - vehicle.VehicleMotion.DistanceTraveled;
+            if (stoppingDistance < vehicle.VehicleMotion.CurrentVelocity.Speed
+                || stoppingDistance >= vehicle.VehicleMotion.CurrentVelocity.Speed - 10)
```

Comment:
This set of conditions confuses me. Actually I'm also confused by why there's a linear relationship between stopping distance and speed, when they presumably have somewhat different units.